processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorSubscription.java

```diff
@@ -76,6 +80,17 @@ public ProcessorSubscription(SubscriptionScope scope,
         setName(String.format("DecatonSubscriptionThread-%s", scope));
     }
 
+    private void onStateChange(SubscriptionStateListener.State newState) {
```

Comment:
since this is not a listener itself, `updateState` would be more appropriate naming?